src/main.c

```diff
@@ -1443,6 +1488,17 @@ static void refloat_thd(void *arg) {
                 d->enable_upside_down = false;
                 d->state.darkride = false;
             }
+
+            if (bms_is_fault_set(d->bms_fault, BMSF_CONNECTION)) {
+                beep_alert(d, 3, true);
+                d->beep_reason = BEEP_BMS_CONNECTION;
+            }
+
+            if (bms_is_fault_set(d->bms_fault, BMSF_CELL_BALANCE)) {
+                beep_alert(d, 3, true);
+                d->beep_reason = BEEP_CELL_BALANCE;
+            }
```

Comment:
IIUC this is gonna be beeping pretty much continuously (not sure the 3 beeps actually make any difference, I'd say not). Did you test this?